cli/src/mesh.rs

```diff
@@ -0,0 +1,49 @@
+use owo_colors::OwoColorize;
+use tokio::time::sleep;
+
+use std::time::Duration;
+
+use utils::mesh::{KaboodlePeer, PeerMetadata};
+
+pub async fn monitor_mesh() -> anyhow::Result<()> {
+    println!("Monitoring mesh ...");
+    let mut mesh = super::peers::create_mesh_peer().await?;
+    let mut discover_rx = mesh
+        .discover_peers()
+        .expect("Unable to get arrivals channel!");
+    let mut depart_rx = mesh
+        .discover_departures()
+        .expect("Unable to get departures channel!");
+
+    loop {
+        if let Some((addr, identity)) = discover_rx.recv().await {
```

Comment:
Should by a `while let` rather than an `if let`? I think as written, it will only read up to one event from each of the `discover_rx` and `depart_rx` channels and then sleep for 5 seconds, which may be problematic on large meshes.